proxy/src/schema.rs

```diff
@@ -53,6 +137,170 @@ impl Query {
         "1.0"
     }
 
+    fn blob(ctx: &Context, id: IdInput, revision: String, path: String) -> FieldResult<Blob> {
+        let repo = GitRepository::new(&ctx.dummy_repo_path).expect("setting up repo failed");
+        let mut browser = GitBrowser::new(&repo).expect("setting up browser for repo failed");
+
+        // Best effort to guess the revision.
+        if let Err(err) = browser
+            .branch(BranchName::new(&revision))
+            .or(browser.commit(Sha1::new(&revision)))
+            .or(browser.tag(TagName::new(&revision)))
+        {
+            let err_fmt = format!("{:?}", err);
+
+            return Err(FieldError::new(
+                "Git error occurred",
+                graphql_value!({ "git": err_fmt }),
+            ));
+        };
+
+        let root = browser
+            .get_directory()
+            .expect("unable to get root directory");
+
+        println!("{:?}", path);
+        let mut p = Path::root();
+        p.append(&mut Path::from_string(&path));
+        let file = root.find_file(&p).expect("unable to find file");
+        let last_commit = browser.last_commit(&p).expect("unable to get last commit");
+        let (_rest, last) = p.split_last();
+
+        Ok(Blob {
+            content: std::str::from_utf8(&file.contents)
+                .expect("invalid content")
+                .to_string(),
+            info: Info {
+                name: last.label,
+                object_type: ObjectType::Blob,
+                last_commit: Commit::from(&last_commit),
+            },
+        })
+    }
+
+    fn commit(ctx: &Context, id: IdInput, sha1: String) -> FieldResult<Option<Commit>> {
```

Comment:
Does it need to be `Option` if we're guaranteed to get it as long as the setting works?

Reply:
Probably not as we will express non-existent errors through the error of `FieldResult`.

Reply:
:+1: